x/tx/decode/decode_test.go

```diff
@@ -112,9 +112,9 @@ func TestDecode(t *testing.T) {
 type dummyAddressCodec struct{}
 
 func (d dummyAddressCodec) StringToBytes(text string) ([]byte, error) {
-	return hex.DecodeString(text)
+	return common.FromHex(text), nil
```

Comment:
Shouldn't matter much with regards to auth but not returning an error may affect the tests for this module.
Though we could interpret an empty byte returned for an invalid hex string.

Reply:
Yeah the thing is that err is ignored when `common.FromHex` is being called here
```
// Hex2Bytes returns the bytes represented by the hexadecimal string str.
func Hex2Bytes(str string) []byte {
	h, _ := hex.DecodeString(str)
	return h
}
```
But still convenient to use it for the `hexPrefix` check. 
So I decided to go that way and leave the check on empty address to the caller.
Anyway, agreed that this should not have impacts on `auth`, and we can think of fixing eventual tests when picking https://polygon.atlassian.net/browse/POS-2455 up.